dbcmetrics/dbcm_logic/dbcplannedmaintenance.py

```diff
@@ -0,0 +1,150 @@
+                        succsessfully_parsed_hours: bool = True
+                    except Exception as e:
+                        logging.warning(f"\tCouldn't parse hours of end of maintenance window, using default hour offset (0 hours).")
+                        succsessfully_parsed_hours: bool = False
+                        start_to_end_window_hours: int = 0
+                    
+                    try:
+                        minute_result = re.findall(r'([\d]+)\s*m', message_string) # returns e.g.: 2, 10, 40 as String in a List
+                        start_to_end_window_min: int = int(minute_result[0])
+                    except Exception as e:
+                        logging.warning(f"\tCouldn't parse minutes of end of maintenance window, using default offset ({self.PLANNNED_MAINTENANCE_DEFAULT_DURATION}min) if parsining hours failed too.")
+                        if(succsessfully_parsed_hours):
+                            # if hours are correctly parse -> dont add default minutes 
+                            start_to_end_window_min: int = 0
+                        else:
+                            start_to_end_window_min: int = self.PLANNNED_MAINTENANCE_DEFAULT_DURATION
+                    
+                    # Create offset timedelta obj for maintenance duration
+                    start_to_end_window_offset = datetime.timedelta(minutes=start_to_end_window_min, hours=start_to_end_window_hours)
+
+                    platform_window_end = platform_window_start + start_to_end_window_offset
+                    
+                    logging.info(f"\tplatform_window_start = {platform_window_start}, platform_window_end = {platform_window_end}")
+                    platform_windows.append([platform_window_start,platform_window_end])
+                
+            
+                # set entry in list
+                self.windows[platform_name] = platform_windows
+                logging.info(f"\tSaved planned maintenance windows for {platform_name}; window count: {len(platform_windows)}")
+
+            except Exception as e:
+                logging.warning(f"Couldn't load/update maintenance window for {platform}")
+                logging.exception(e)
+
+    def refresh_metrics(self):
+        for platform_name, plattform_windows in self.windows.items():
+            in_window = False
+            for platform_window_start, platform_window_end  in plattform_windows:
+                if self.currently_in_window(platform_window_start, platform_window_end):
+                    in_window = True
+                    continue
+            # Set metric
+            if in_window:
+                self.metric.labels(platform=platform_name).set(1)
+            else:
+                self.metric.labels(platform=platform_name).set(0)
+
+
+    @staticmethod
+    def currently_in_window(platform_window_start: datetime.datetime, platform_window_end: datetime.datetime) -> bool:
+
+        tz_berlin = pytz.timezone('Europe/Berlin')
```

Comment:
It's not a good practice to hardcode the used timezone

Reply:
Has now been improved